decentraland/unity-explorer
# Feat: memory budgeting for streamable assets

# Changes
### Functional
- added MemoryBudgetProvider with respective ISystemMemory
- used MemoryBudgetProvider to lock loading of streamable assets in DefferedLoadingSystem when memory is full
- add visualisation of used memory in the debug window
- add DCL debug MenuItem with the menu to simulate Full, Warninig and Normal memory states

### Structural
- marked Addressables groups as static for better optimization. Non-static is useful only for remote groups.
- created PerformanceAndDiagnostics folder where I moved Profiling, Diagnostics, Debug and PerformanceBudgeting; + adjusted namespaces
- restructered asmdef and asmref for PerformanceBudgeting and Profiling.

# How to test
1. Run the scene
2. Select realm
3. When it started to load some assets, go to DCL Tools -> Memory Budgeting -> FULL to simulate full memory
4. Observe assets stopped to load
5. Select DCL Tools -> Memory Budgeting -> NORMAL (or WARNING) and observe remained assets being loaded
